src/os/bluestore/bluefs_types.h

```diff
@@ -111,11 +117,11 @@ struct bluefs_fnode_t {
   template<typename T, typename P>
   friend std::enable_if_t<std::is_same_v<bluefs_fnode_t, std::remove_const_t<T>>>
   _denc_friend(T& v, P& p) {
-    DENC_START(1, 1, p);
+    DENC_START(2, 2, p);
     denc_varint(v.ino, p);
     denc_varint(v.size, p);
     denc(v.mtime, p);
-    denc(v.__unused__, p);
+    denc(v.type, p);
```

Comment:
For wal_v2 files one could store both size (bluefs file size) and wal_size (size of respective wal after stripping envelopes).

Reply:
That way, checking for truncated content can start from the saved size.

Reply:
Plus, an additional bit signalling that if file is now closed could skip check altogether.
But I am not sure if its worth it, since "check ahead" function has to be created anyway.